input-explainer.md

```diff
@@ -280,6 +292,91 @@ When the canvas receives a `pointerdown` event an `XRInputSource` is created wit
 
 For each of these events the `XRInputSource`'s target ray must be updated to originate at the point that was interacted with on the canvas, projected onto the near clipping plane (defined by the `depthNear` attribute of the `XRSession`) and extending out into the scene along that projected vector.
 
+## Button and Axis State
+
+Some applications need more than point-and-click style interaction provided by the `select` events. For devices that make use of controllers with buttons and axes, more complete information about the state of those inputs can be observed via the `XRInputSource`'s `gamepad` attribute.
+
+If the `XRInputState` represents an XR controller device with buttons or axes the `gamepad` attribute is an instance of a [`Gamepad`](https://w3c.github.io/gamepad/#gamepad-interface) interface, and `null` otherwise. Examples of controllers that may expose their state this way include Oculus Touch, Vive wands, Oculus Go and Daydream controllers, or other similar devices. Controllers not directly associated with the XR device, such as the majority of traditional gamepads, should not be exposed using this interface. Tracked devices without discreet inputs, such as optical hand tracking, should also have a `null` `gamepad` attribute.
```

Comment:
> should not be exposed using this interface.

I've come around to agree this should say "must not"

Reply:
And at the very least, if we do report a traditional gamepad here, it should not be present in the `navigator.getGamepads()` array at the same time.

Reply:
> And at the very least, if we do report a traditional gamepad here, it should not be present in the `navigator.getGamepads()` array at the same time.

Bingo: It's less that we need to be super-prescriptive about where you enumerate a given device than that we ensure devices are not enumerated in two different APIs, creating an unbounded deduplication requirement that we offload to apps.